test/test_utils.py

```diff
@@ -989,3 +991,118 @@ def test_get_process_names(self):
         res = get_process_names()
         res = list(res)
         self.assertEquals(res, [fake_process_name], "Expected an empty list, Actual: %s" % res)
+
+
+class TestIsInsightsRegisterEnabled(fixture.SubManFixture):
+    """
+    This is a group of tests for the is_insights_register_enabled method.
+    As the method depends on "check_returncode", it does not test the functionality of that.
+    """
+    def setUp(self):
+        check_returncode_patch = patch('subscription_manager.utils.check_returncode')
+        self.check_returncode_mock = check_returncode_patch.start()
+        self.addCleanup(check_returncode_patch.stop)
+
+    def test_is_insights_register_enabled(self):
+        self.check_returncode_mock.return_value = 0
+        self.assertTrue(is_insights_register_enabled())
+        self._assert_called_correct_cmd()
+
+    def test_is_insights_register_disabled(self):
+        self.check_returncode_mock.return_value = 1
+        self.assertFalse(is_insights_register_enabled())
+        self._assert_called_correct_cmd()
+
+    def _assert_called_correct_cmd(self):
+        self.check_returncode_mock.assert_called_with('systemctl is-enabled insights-register.path')
+
+
+class TestIsInsightsInstalled(fixture.SubManFixture):
+    """
+    This is a group of tests for the is_insights_installed method.
+    As the method depends on "check_returncode", it does not test the functionality of that.
+    """
+
+    def setUp(self):
+        check_returncode_patch = patch('subscription_manager.utils.check_returncode')
+        self.check_returncode_mock = check_returncode_patch.start()
+        self.addCleanup(check_returncode_patch.stop)
+
+    def test_is_insights_register_enabled(self):
+        self.check_returncode_mock.return_value = 0
+        self.assertTrue(is_insights_installed())
+        self._assert_called_correct_cmd()
+
+    def test_is_insights_register_disabled(self):
```

Comment:
Similar issue here. I would change the name of the method to something like: `test_is_insights_client_not_installed`